core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -242,10 +242,28 @@ private FormValidation verifySignature(JSONObject o) throws IOException {
     /**
      * Let sub-classes of UpdateSite provide their own signature validator.
      * @return the signature validator.
+     * @deprecated use {@link #getJsonSignatureValidator(@CheckForNull String)} instead.
      */
+    @Deprecated
     @Nonnull
     protected JSONSignatureValidator getJsonSignatureValidator() {
-        return new JSONSignatureValidator("update site '"+id+"'");
+        return getJsonSignatureValidator(null);
+    }
+
+    /**
+     * Let sub-classes of UpdateSite provide their own signature validator.
+     * @param name, the name for the JSON signature Validator object.
+     *              if name is null, then the default name will be used,
+     *              which is "update site" followed by the update site id
+     * @return the signature validator.
+     * @since 2.15
+     */
+    @Nonnull
+    protected JSONSignatureValidator getJsonSignatureValidator(@CheckForNull String name) {
+        if (name == null) {
+            name = "update site '" + id + "'";
```

Comment:
This string seems like magic that has a tendency to be copied and pasted all over the place. Is there a better solution?